File: views/src/main/java/com/mtime/base/widget/OnViewClickListener.java
package com.mtime.base.widget;

import android.view.View;

import com.mtime.base.utils.MViewUtil;

/**
 * Created by ZhouSuQiang on 2017/11/25.
 * OnClickListener的包装类，防止重复点击
 */

public abstract class OnViewClickListener implements View.OnClickListener {
    
    @Override
    public void onClick(View v) {
        if(MViewUtil.isNotRepeatClick(v))
            onClicked(v);
    }
    
    public abstract void onClicked(View v);
}

File: widget/src/main/java/com/kotlin/android/widget/adapter/multitype/adapter/callback/OnViewClickListener.kt
package com.kotlin.android.widget.adapter.multitype.adapter.callback

import android.view.View

/**
 * 创建者: zl
 * 创建时间: 2020/7/1 10:06 AM
 * 描述:
 */
interface OnViewClickListener {
    // 不需要额外参数事件时，默认转发给带额外参数事件
    fun onClick(view: View) {
        onClick(view, null)
    }

    fun onClick(view: View, any: Any?) {

    }
}